integration/test/performance_tracker/performance_query_utils.go

```diff
@@ -0,0 +1,143 @@
+package data_collector
+
+import (
+	"time"
+	"errors"
+	"context"
+	"encoding/json"
+	"os"
+	"fmt"
+	
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+const (
+	Namespace = "CWAgent"
+	DimensionName = "InstanceId"
+	Stat = "Average"
+	Period = 30
+)
+// GenerateGetMetricInputStruct generates the struct required to make a query request to cloudwatch's GetMetrics
+func GenerateGetMetricInputStruct(ids, metricNames []string, instanceId string, timeDiff int) (*cloudwatch.GetMetricDataInput, error) {
+	if len(ids) != len(metricNames) {
+		return nil, errors.New("Mismatching lengths of metric ids and metricNames")
+	}
+	
+	if len(ids) == 0 || len(metricNames) == 0 || instanceId == "" || timeDiff == 0 {
+		return nil, errors.New("Must supply metric ids, metric names, instance id, and time to collect metrics")
+	}
+
+	dimensionValue := instanceId
+	metricDataQueries := []types.MetricDataQuery{}
+	
+	//generate list of individual metric requests
+	for i, id := range ids {
+		metricDataQueries = append(metricDataQueries, ConstructMetricDataQuery(id, Namespace, DimensionName, dimensionValue, metricNames[i], timeDiff))
+	}
+	
+	timeNow := time.Now()
+	input := &cloudwatch.GetMetricDataInput{
+		EndTime:   aws.Time(time.Unix(timeNow.Unix(), 0)),
+		StartTime: aws.Time(time.Unix(timeNow.Add(time.Duration(-timeDiff)*time.Minute).Unix(), 0)),
+		MetricDataQueries: metricDataQueries,
+	}
+
+	return input, nil
+}
+
+// ConstructMetricDataQuery is a helper function for GenerateGetMetricInputStruct and constructs individual metric requests
+func ConstructMetricDataQuery(id, namespace, dimensionName, dimensionValue, metricName string, timeDiff int) (types.MetricDataQuery) {
+	query := types.MetricDataQuery{
+		Id: aws.String(id),
+		MetricStat: &types.MetricStat{
+			Metric: &types.Metric{
+				Namespace:  aws.String(namespace),
+				MetricName: aws.String(metricName),
+				Dimensions: []types.Dimension{
+					types.Dimension{
+						Name:  aws.String(dimensionName),
+						Value: aws.String(dimensionValue),
+					},
+				},
+			},
+			Period: aws.Int32(int32(Period)),
+			Stat:   aws.String(Stat),
+		},
+	}
+
+	return query
+}
+
+func GetPerformanceMetrics(instanceId string, agentRuntime int) ([]byte, error) {
```

Comment:
Any strong reason why we return a byte array here?

Reply:
It's returning a byte array because that is what json.MarshalIndent returns and I believe that is what the database api part takes as an initial input